dashboard/modules/serve/serve_agent.py

```diff
@@ -53,7 +53,13 @@ async def get_all_deployments(self, req: Request) -> Response:
         if client is None:
             config = ServeApplicationSchema.get_empty_schema_dict()
         else:
-            config = client.get_app_config()
+            try:
+                config = client.get_app_config()
```

Comment:
@edoakes should we keep an in memory copy for the data so that we don't need gcs to retrieve them?